client/src/components/groups/GroupComponent.svelte

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+    import { push } from "svelte-spa-router";
+    import { onDestroy, onMount } from "svelte";
+
+    import RoundedButton from "../UI/buttons/RoundedButton.svelte";
+    import ToggleButton from "../UI/buttons/ToggleButton.svelte";
+    import type { Group } from "../../../types/groupType";
+    import Modal from "../UI/modal/Modal.svelte";
+    import GroupService from "../../services/groupService";
+    import { tr } from "../../utils/i18nHelper";
+    import { socketManager } from "../../managers/socketManager";
+    import EntityService from "../../services/entityService";
+    import type { Entity } from "../../../types/entityType";
+    import {
+        toggle
+    } from '../../services/entities/generalService';
+
+    export let group: Group;
+    let checked = group.state === "on";
+
+    function handleToggle() {
+        console.log(group.groupId);
+        if (!group.groupId){
+            return;
+        }
+        toggle(`group.${group.groupId}`);
+    }
+    
+    function groupEntityUpdatedHandler(data: any) {
+        checked = data?.state === "on";
+    }
+    function groupUpdatedHandler(data: any) {
+        if (data.groupId && data.groupId === group.groupId){
+            group = data;
+            group = GroupService.updateGroupNameIfIsImplicit(group);  
+        }
+    }
+
+    onMount(async () => {
+        if (!group.groupId) {
+            return;
+        }
+        socketManager.registerListenerById<Entity<any>>(
```

Comment:
Isn't this a group ?
```suggestion
        socketManager.registerListenerById<Entity<Group>>(
```